src/processTargets/targetUtil/insertionRemovalBehaviors/TokenInsertionRemovalBehavior.ts

```diff
@@ -59,45 +58,81 @@ export function getTokenTrailingDelimiterTarget(
  * removal range is designed to be used with things that should clean themselves
  * up as if they're a range of tokens.
  * @param target The target to get the token removal range for
- * @param contentRange Can be used to override the content range instead of
- * using the one on the target
  * @returns The removal range for the given target
  */
-export function getTokenRemovalRange(
-  target: Target,
-  contentRange?: Range
-): Range {
-  const { document } = target.editor;
-  const actualContentRange = contentRange ?? target.contentRange;
-  const removalRange = getDelimitedSequenceRemovalRange(target, contentRange);
-
-  if (!actualContentRange.isEqual(removalRange)) {
-    const fullRange = expandToFullLine(target.editor, actualContentRange);
-    const fullText = document.getText(fullRange);
-    const fullTextOffset = document.offsetAt(fullRange.start);
-
-    const numTokensContentRangeRemoved = calculateNumberOfTokensAfterRemoval(
-      document,
-      fullText,
-      fullTextOffset,
-      actualContentRange
-    );
-
-    const numTokensRemovalRangeRemoved = calculateNumberOfTokensAfterRemoval(
-      document,
-      fullText,
-      fullTextOffset,
-      removalRange
-    );
-
-    // Using removal range has not merged any tokens. Removal range is ok to use.
-    if (numTokensContentRangeRemoved === numTokensRemovalRangeRemoved) {
-      return removalRange;
+export function getTokenRemovalRange(target: Target): Range {
```

Comment:
I decided to rework this function, because I think that `getDelimitedSequenceRemovalRange` was obscuring the full picture here, and we were really only relying on it for the convention of preferring trailing over leading.  We do lose the fact that we have a single source of truth for that convention, but given how special this token removal code has become I don't think it was worth working around that function